components/eam/src/chemistry/mozart/mo_photo.F90

```diff
@@ -1212,103 +668,92 @@ subroutine cloud_mod( zen_angle, clouds, lwc, delp, srf_alb, &
     !---------------------------------------------------------
     where( clouds(:) /= 0._r8 )
-       del_lwp(:) = rgrav * lwc(:) * delp(:) * 1.e3_r8 / clouds(:)
+       del_lwp(:) = rgrav * lwc(:) * delp(:) * 1.e3_r8 / clouds(:) ! the unit is (likely) g/m^2
+       del_tau(:) = del_lwp(:) * f_lwp2tau * clouds(:)**1.5_r8
     elsewhere
        del_lwp(:) = 0._r8
-    endwhere
-    !---------------------------------------------------------
-    !    	... form tau for each model layer
-    !---------------------------------------------------------
-    where( clouds(:) /= 0._r8 )
-       del_tau(:) = del_lwp(:) *.155_r8 * clouds(:)**1.5_r8
-    elsewhere
        del_tau(:) = 0._r8
-    end where
+    endwhere
     !---------------------------------------------------------
-    !    	... form integrated tau from top down
+    !    	... form integrated tau and cloud cover from top down
     !---------------------------------------------------------
     above_tau(1) = 0._r8
-    do k = 1,pverm
-       above_tau(k+1) = del_tau(k) + above_tau(k)
-    end do
+    above_cld(1) = 0._r8
+    do kk = 1,pverm
+       above_tau(kk+1) = del_tau(kk) + above_tau(kk)
+       above_cld(kk+1) = clouds(kk) * del_tau(kk) + above_cld(kk)
+    enddo
+    do kk = 2,pver
+       if( above_tau(kk) /= 0._r8 ) then
+          above_cld(kk) = above_cld(kk) / above_tau(kk)
+       else
+          above_cld(kk) = above_cld(kk-1)
+       endif
+    enddo
     !---------------------------------------------------------
-    !    	... form integrated tau from bottom up
+    !    	... form integrated tau and cloud cover from bottom up
     !---------------------------------------------------------
     below_tau(pver) = 0._r8
-    do k = pverm,1,-1
-       below_tau(k) = del_tau(k+1) + below_tau(k+1)
-    end do
-    !---------------------------------------------------------
-    !	... form vertically averaged cloud cover above and below
-    !---------------------------------------------------------
-    above_cld(1) = 0._r8
-    do k = 1,pverm
-       above_cld(k+1) = clouds(k) * del_tau(k) + above_cld(k)
-    end do
-    do k = 2,pver
-       if( above_tau(k) /= 0._r8 ) then
-          above_cld(k) = above_cld(k) / above_tau(k)
-       else
-          above_cld(k) = above_cld(k-1)
-       end if
-    end do
     below_cld(pver) = 0._r8
-    do k = pverm,1,-1
-       below_cld(k) = clouds(k+1) * del_tau(k+1) + below_cld(k+1)
+    do kk = pverm,1,-1
+       below_tau(kk) = del_tau(kk+1) + below_tau(kk+1)
+       below_cld(kk) = clouds(kk+1) * del_tau(kk+1) + below_cld(kk+1)
     end do
-    do k = pverm,1,-1
-       if( below_tau(k) /= 0._r8 ) then
-          below_cld(k) = below_cld(k) / below_tau(k)
+    do kk = pverm,1,-1
+       if( below_tau(kk) /= 0._r8 ) then
+          below_cld(kk) = below_cld(kk) / below_tau(kk)
        else
-          below_cld(k) = below_cld(k+1)
-       end if
-    end do
+          below_cld(kk) = below_cld(kk+1)
+       endif
+    enddo
```

Comment:
These above and below tau/cld calculations are very similar. Can we factor these out in a subroutine?

Reply:
I can separated them out in a subroutine, but can't think out a way to simplify the code.

Reply:
You are right. The indices are inconsistent which makes it harder to simply. Ignore my comment. It is okay as is.